### PR TITLE
e2e: Add conformance test to verify cluster operators have versions

### PR DESCRIPTION
**Pull request by smarterclayton**
All operators must report an operator version that matches the cluster
release version for the cluster to pass.

### CODE REVIEW

**Comment by smarterclayton** [comment]
@abhinavdahiya @derekwaynecarr

**Comment by smarterclayton** [comment]
Pull https://github.com/openshift/cluster-image-registry-operator/pull/219

File: test/extended/operators/operators.go [hunk 1]
```diff
@@ -167,6 +169,37 @@ var _ = g.Describe("[Feature:Platform][Smoke] Managed cluster should", func() {
 	})
 })
 
+var _ = g.Describe("[Feature:Platform] Managed cluster should", func() {
+	defer g.GinkgoRecover()
+
+	g.It("have operators on the cluster version", func() {
```
Comment (abhinavdahiya) [inline test/extended/operators/operators.go]: `s/version/operator`

Comment (smarterclayton) [reply test/extended/operators/operators.go]: ?  cluster version is the version we want to have operator report

Comment (abhinavdahiya) [reply test/extended/operators/operators.go]: the test reads `[Feature:Platform] Managed cluster should have operators on the cluster version` which threw me off a little... this is fine as-is

**Comment by smarterclayton** [comment]
Correctly fails:

```
[]v1.OperandVersion{v1.OperandVersion{Name:"cluster-autoscaler", Version:"v0.0.0-62-g73c4665-dirty"}}
Expected
    <string>: cluster-autoscaler
to equal
    <string>: operator
```

**Comment by derekwaynecarr** [comment]
this makes sense, will start pushing across teams.

**Comment by runcom** [comment]
is https://github.com/openshift/cluster-image-registry-operator/pull/225/commits/473e60bd4d845d11bc3fe09391a312c77a2cb9cb basically the _template_ to use to implement this? The MCO currently reports its versions and its operands (for the operands we report the pullspec from the release payload)

**Comment by smarterclayton** [comment]
Yes, it is.

All components must report the first version:

```
name: operator
value: 0.0.1-snapshot
```

Components should not report versions beyond that that are distinctly different without approval.  We haven't sorted out RHCoS version reporting, so I'd recommend removing that and queueing up a card where we go through what you have to do (you need a component, I have to approve the component name, we need the machine-os-content images to be labelled a certain way, etc).

**Comment by runcom** [comment]
> Components should not report versions beyond that that are distinctly different without approval. We haven't sorted out RHCoS version reporting, so I'd recommend removing that and queueing up a card where we go through what you have to do (you need a component, I have to approve the component name, we need the machine-os-content images to be labelled a certain way, etc).

alrighty, we aren't reporting the rhcos version anywhere (but MCD logs) - but yeah, will create a card for that

**Comment by smarterclayton** [comment]
RHCoS version is reported by nodes which is ok for right now.  It will go into the payload so it shows up in changelogs just not there yet.

**Comment by smarterclayton** [comment]
/retest

should pass now

**Pull request merged**